### PR TITLE
Release App Service

### PR DESCRIPTION
**Pull request by jsondoo**
This moves App Service card out of preview mode (and Functions back into preview mode)

Should be merged after #1077 and #1068 
image

### CODE REVIEW

**Review by jcoc611-microsoft — DISMISSED** [review]

**Comment by SahilTara** [comment]
Just wondering why move functions to preview? Did something change?

**Comment by Tanya0609** [comment]
> Just wondering why move functions to preview? Did something change?

This is a decision made by the entire team and consulted with Azure Service advocate. We are temporarily move Azure functions into Preview mode, and it should be brought back once the user flow gets improved. One potential change need to be made is: utilizing the Azure Functions extension. 

Here are the reasons why:
1.	The user flow of using Azure functions is not appropriate.
-	It makes more sense for people to choose Azure Functions as their backend, but we don’t have the capacity to implement that this summer.
2.	The Azure Functions are not properly hooked up with generated project code.
-	At the moment, we only create a Hello World application when we create the Azure Functions. We don’t have any Functions hooked up properly with our generated web application. When we fleshed out the user flow for how to actually attach Azure Functions, it requires a lot of effort for them to figure out how to connect it to their project. At that point, they might as well have gone through the documentation themselves to understand how to use it (especially since a lot of the user feedback we got from cloud developer advocates is "why do I need to choose the # of functions I need right now?"). With the introduction of App Service, it's superior to Functions in almost every single way since it handles everything for the user. 
3. Only 1 or 2 out of 140ish people are deploying Azure Functions via WebTS weekly now.

**The goal is not remove Functions, but to improve the user experience.** Functions is the trend. We want to have a nice Azure Functions experience. We are investigating the Azure Function extension.

**Review by ali-kha — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ Web Template Studio approaches full-stack web app creation using the following t
...
 - **App pages**: Next, to accelerate app creation, we provide a number of app page templates that you can use to add common UI pages into your new app. The current page templates include: _blank page_, common layouts (*e.g., master detail) and pages that implement common patterns (*e.g., grid, list). Using the wizard, add as many of the pages as you need, providing a name for each one, and we'll generate them for you.
-- **Cloud Services**: Lastly, you specify which Azure cloud services you want to use, and we'll build out the framework for the services into your app including tagging 'TODO' items. Currently supported services cover storage (_Azure Cosmos DB_), and compute (_Azure Functions_).
+- **Cloud Services**: Lastly, you specify which Azure cloud services you want to use, and we'll build out the framework for the services into your app including tagging 'TODO' items. Currently supported services cover storage (_Azure Cosmos DB_), and compute (_Azure App Service_).
```
Comment (jasmingoh) [inline README.md]: we are handling the read me changes in another pr, is it ok if you just revert this change?

File: src/extension/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Web Template Studio approaches full-stack web app creation using the following t
...
 - **App pages**: Next, to accelerate app creation, we provide a number of app page templates that you can use to add common UI pages into your new app. The current page templates include: _blank page_, common layouts (*e.g., master detail) and pages that implement common patterns (*e.g., grid, list). Using the wizard, add as many of the pages as you need, providing a name for each one, and we'll generate them for you.
-- **Cloud Services**: Lastly, you specify which Azure cloud services you want to use, and we'll build out the framework for the services into your app including tagging 'TODO' items. Currently supported services cover storage (_Azure Cosmos DB_), and compute (_Azure Functions_).
+- **Cloud Services**: Lastly, you specify which Azure cloud services you want to use, and we'll build out the framework for the services into your app including tagging 'TODO' items. Currently supported services cover storage (_Azure Cosmos DB_), and compute (_Azure App Service_).
```
Comment (jasmingoh) [inline src/extension/README.md]: here as well

**Review by ali-kha — APPROVED** [review]

**Review by jasmingoh — APPROVED** [review]

File: src/client/src/containers/AzureSubscriptions/index.tsx [hunk 3]
```diff
@@ -235,6 +235,15 @@ class AzureSubscriptions extends React.Component<Props, IState> {
...
     const uniqueServiceTypes = [...new Set(serviceTypes)];
+
+    let numHostingServiceCards = 0;
+    azureServiceOptions.forEach(serviceOption => {
+      const isCardShown = isPreview || !serviceOption.isPreview;
+      if (serviceOption.type === servicesEnum.HOSTING && isCardShown) {
+        numHostingServiceCards++;
+      }
+    });
+
```
Comment (jasmingoh) [inline src/client/src/containers/AzureSubscriptions/index.tsx:246]: wait. what is this..

Comment (jsondoo) [reply src/client/src/containers/AzureSubscriptions/index.tsx:246]: this is logic so that the "You can only choose one hosting service" message only shows up if there's more than 1 hosting services card

Comment (jsondoo) [reply src/client/src/containers/AzureSubscriptions/index.tsx:246]: also for preview mode too (it will have 2 hosting services cards)

**Review by streetcoder123 — APPROVED** [review]

**Review by piotrmark — APPROVED** [review]

**Review by annielovesjs — APPROVED** [review]
fantastic!
